### ISSUE TITLE
SC2143 false positive

### ISSUE THREAD

**Issue by dgutson** [koalaman/shellcheck#2198]
#### For bugs
- Rule Id: SC2143
- My shellcheck version: "online" 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
if [[ "$lsb_dist" == "rhel" ]] && [[ ! $(grep -q "VERSION_ID=\"7" /etc/os-release) ]]; then
	sudo sed -i 's/$releasever/7/' /etc/yum.repos.d/docker-ce.repo
fi

```

#### Here's what shellcheck currently says:
![image](https://user-images.githubusercontent.com/9671770/113727948-023d5200-96cc-11eb-9916-cae2eb004b47.png)

#### Here's what I wanted or expected to see:
No warning about grep -q, since the script already has that option.

**Comment by koalaman** [comment]
The warning is indeed confusing, but it does indicate a real problem. You should not use `[[ $(grep -q ..) ]]` because it has no output, and therefore will always be false. You should remove the brackets:

```
if [[ "$lsb_dist" == "rhel" ]] && ! $(grep -q "VERSION_ID=\"7" /etc/os-release); then
	sudo sed -i 's/$releasever/7/' /etc/yum.repos.d/docker-ce.repo
fi
```

This might deserve its own warning

**Comment by dgutson** [comment]
Thanks for the fix of our code :)
I agree that the message is misleading, pls consider to turn this ```[[]]``` issue in as a separate warning.

**Comment by dxkrs** [comment]
Maybe, I am meeting this promble too! I have tried this code on the website https://www.shellcheck.net/
```
#!/bin/sh

_pwd='a1234'
if [ "$(echo "$_pwd" | grep -q -E "^[a-zA-Z]{1}$")" ];then
        echo 'oh no!'
fi
```
the website show the SC2143 style promble, I don't know how to match this style.
